code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -49,12 +49,30 @@
 		"bromine",
 		"stable_plasma"
 	)
+//these become available once upgraded.
+	var/list/upgrade_reagents = list(
+		"oil",
+		"ammonia"
+	)
+
+	var/list/upgrade_reagents2 = list(
+		"acetone",
+		"phenol",
+		"diethylamine"
+	)
+
+	var/list/upgrade_reagents3 = list(
+		"omnizine",
+		"glycerol"
+	)
+
 	var/list/emagged_reagents = list(
 		"space_drugs",
-		"morphine",
+		"sulfonal",
+		"frostoil",
 		"carpotoxin",
-		"mine_salve",
-		"toxin"
```

Comment:
this is actually useful, add it to the regular chem dispenser if you want but definitely do not remove